sovrin_node/server/domain_req_handler.py

```diff
@@ -0,0 +1,346 @@
+import json
+from binascii import unhexlify
+from typing import List
+from hashlib import sha256
+
+from plenum.common.exceptions import InvalidClientRequest, \
+    UnauthorizedClientRequest, UnknownIdentifier
+from plenum.common.constants import TXN_TYPE, TARGET_NYM, RAW, ENC, HASH, \
+    VERKEY, DATA, NAME, VERSION
+from plenum.common.types import f
+from plenum.common.util import check_endpoint_valid
+from plenum.server.domain_req_handler import DomainRequestHandler as PHandler
+from sovrin_common.auth import Authoriser
+from sovrin_common.constants import NYM, ROLE, ATTRIB, ENDPOINT, SCHEMA, \
+    ISSUER_KEY, REF
+from sovrin_common.types import Request
+from stp_core.common.log import getlogger
+from stp_core.network.exceptions import EndpointException
+
+
+logger = getlogger()
+
+
+class DomainReqHandler(PHandler):
+    MARKER_ATTR = "\01"
+    MARKER_SCHEMA = "\02"
+    MARKER_IPK = "\03"
+    LAST_SEQ_NO = "lsn"
+    VALUE = "val"
+
+    def __init__(self, ledger, state, requestProcessor, idrCache, attributeStore):
+        super().__init__(ledger, state, requestProcessor)
+        self.idrCache = idrCache
+        self.attributeStore = attributeStore
+
+    def onBatchCreated(self, stateRoot):
+        self.idrCache.currentBatchCreated(stateRoot)
+
+    def onBatchRejected(self, stateRoot=None):
+        self.idrCache.batchRejected(stateRoot)
+
+    def updateState(self, txns, isCommitted=False):
+        for txn in txns:
+            typ = txn.get(TXN_TYPE)
+            nym = txn.get(TARGET_NYM)
+            if typ == NYM:
+                data = {f.IDENTIFIER.nm: txn.get(f.IDENTIFIER.nm)}
+                if ROLE in txn:
+                    data[ROLE] = txn.get(ROLE)
+                if VERKEY in txn:
+                    data[VERKEY] = txn.get(VERKEY)
+                self.updateNym(nym, data, isCommitted=isCommitted)
+            elif typ == ATTRIB:
+                self._addAttr(txn)
+            elif typ == SCHEMA:
+                self._addSchema(txn)
+            elif typ == ISSUER_KEY:
+                self._addIssuerKey(txn)
+            else:
+                logger.debug('Cannot apply request of type {} to state'.format(typ))
+
+    def commit(self, txnCount, stateRoot, txnRoot) -> List:
+        r = super().commit(txnCount, stateRoot, txnRoot)
+        self.idrCache.onBatchCommitted(unhexlify(stateRoot.encode()))
+        return r
+
+    def canNymRequestBeProcessed(self, identifier, msg) -> (bool, str):
+        nym = msg.get(TARGET_NYM)
+        if self.hasNym(nym, isCommitted=False):
+            if not self.idrCache.hasTrustee(identifier, isCommitted=False) and \
+                            self.idrCache.getOwnerFor(nym, isCommitted=False) != identifier:
+                reason = '{} is neither Trustee nor owner of {}'\
+                    .format(identifier, nym)
+                return False, reason
+        return True, ''
+
+    def doStaticValidation(self, identifier, reqId, operation):
```

Comment:
Please split the method to perform static validation for each txn type (like `doStaticValidationNym`, `doStaticValidationAttr`)